File: Easy/3461. Check If Digits Are Equal in String After Operations I/Solution.cs
using System.Text;

public class Solution {
    public bool HasSameDigits(string s) {
        
        while (s.Length > 2) {
            // Use StringBuilder for performance
            StringBuilder newS = new StringBuilder();
            
            for (int i = 0; i < s.Length - 1; i++) {
                // Convert char to int
                int num1 = s[i] - '0';
                int num2 = s[i+1] - '0';
                
                int newNum = (num1 + num2) % 10;
                newS.Append(newNum);
            }
            // Update 's' for the next loop
            s = newS.ToString();
        }
        
        // Final check
        return s[0] == s[1];
    }
}

File: Easy/3461. Check If Digits Are Equal in String After Operations I/Solution.java
import java.lang.StringBuilder;

class Solution {
    public boolean hasSameDigits(String s) {

        while (s.length() > 2) {
            // Create temp new string
            StringBuilder newS = new StringBuilder();

            for (int i = 0; i < s.length() - 1; i++) {
                // Convert char to int
                int num1 = s.charAt(i) - '0';
                int num2 = s.charAt(i+1) - '0';

                int newNum = (num1 + num2) % 10;
                newS.append(newNum);
            }
            // Update 's' for the next loop
            s = newS.toString();
        }

        // Final check
        return s.charAt(0) == s.charAt(1);
    }
}